src/librustc_borrowck/borrowck/mod.rs

```diff
@@ -1208,14 +1208,17 @@ impl<'a, 'tcx> BorrowckCtxt<'a, 'tcx> {
                 let let_span = self.tcx.hir.span(node_id);
                 match self.local_binding_mode(node_id) {
                     ty::BindByReference(..) => {
-                        let snippet = self.tcx.sess.codemap().span_to_snippet(let_span);
-                        if let Ok(snippet) = snippet {
+                        if let Ok(snippet) = self.tcx.sess.codemap().span_to_snippet(let_span) {
+                            let replace_str = if snippet.starts_with("ref ") {
+                                snippet.replacen("ref ", "ref mut ", 1)
+                            } else {
+                                snippet
+                            };
                             db.span_label(
```

Comment:
Could we change this to be a `span_suggestion`? That way editors will know they can perform the replacement and the following code:

```
                            let replace_str = if snippet.starts_with("ref ") {
                                snippet.replacen("ref ", "ref mut ", 1)
                            } else {
                                snippet
                            };
                            db.span_label(
                                let_span,
                                "use a mutable reference instead",
                                replace_str,
                            );
```

output would look like this:

```
error[E0594]: cannot assign to immutable borrowed content `*my_ref`
  --> $DIR/issue-51244.rs:13:5
   |
LL |     let ref my_ref @ _ = 0;
   |         -------------- use a mutable reference instead: `ref mut my_ref @ _`
LL |     *my_ref = 0;
   |     ^^^^^^^^^^^ cannot borrow as mutable
```

This will likely require you to run `./x.py test src/test/ui --bless` to update the output in other test files too.

If you do not wish to make this change now, please file a ticket to follow up on all suggestions on this file, so that the team doesn't forget to address this later.